### ISSUE TITLE
User variable not taken into account

### ISSUE THREAD

**Issue by martinm82** [hashicorp/packer#9090]
#### Overview of the Issue

Given the following setup:

* Common vars file (vars/all.json)
* Region specific vars file (eg. vars/eu-west-1.json)
* build.json using vars from files above

When running 

```
docker run --rm -it -v `pwd`:`pwd` -w `pwd` hashicorp/packer:1.5.5 build -var-file=vars/all.json -var-file=vars/eu-west-1.json -var 'aws_access_key=AKIxxxxxx' -var 'aws_secret_key=xxxxx' build.json
```

I get the following error:

```
Build 'amazon-ebs' errored: error validating regions: AuthFailure: Credential must have exactly 5 slash-delimited elements, e.g. keyid/date/region/service/term, got
```

This issue does not occur when using, 1.4.1, 1.4.2, 1.5.0, 1.5.4

#### Reproduction Steps

Create a vars file with 

```
{
  "aws_access_key": "PASS THE KEY WITH -var 'aws_access_key=<YOUR_KEY>'",
  "aws_secret_key": "PASS THE KEY WITH -var 'aws_secret_key=<YOUR_KEY>'",
}
```

and a build.json with 

```
{
  "builders": [{
    "access_key": "{{user `aws_access_key`}}",
    "secret_key": "{{user `aws_secret_key`}}",
[...]
```

### Packer version

I did not went through old version but it seems the problem started occurring with the latest version 1.5.5.

### Simplified Packer Buildfile

https://gist.github.com/martinm82/b3360dc5fc29e70784d7b9619bce8d59

### Operating system and Environment details

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

### Log Fragments and crash.log files

See the log file attached to the gist: https://gist.github.com/martinm82/b3360dc5fc29e70784d7b9619bce8d59

**Comment by nywilken** [comment]
Hi @martinm82 thanks for reaching out. We will take a look into this issue for the next release.

**Comment by nywilken** [comment]
Hello again, apologies for not mentioning earlier. I believe this has already been resolved in https://github.com/hashicorp/packer/issues/8961 and is set to be released in the release v1.5.6. Can you check if you are still able to reproduce this issue using a nightly release https://github.com/hashicorp/packer/releases/tag/nightly

**Comment by martinm82** [comment]
@nywilken Yes, seems that this is indeed a duplicate. I will check tomorrow with the nightly build
